### PR TITLE
Add failure messages.

### PR DESCRIPTION
**Pull request by BigBlueHat**
Working my way through the full list of asserts/shoulds. The goal is to provide meaningful/actionable output to implementers.

Related to this PR improving error display: https://github.com/digitalbazaar/mocha-w3c-interop-reporter/pull/40

Would love early feedback if you can spare the cycles.

### CODE REVIEW

**Comment by PatStLouis** [comment]
I like this and I'll keep this in mind moving forward!

File: tests/10-vcdm2.js [hunk 1]
```diff
@@ -97,21 +101,27 @@ describe('Contexts', function() {
             presentation: require('./input/presentation-ok.json'),
             options: createOptions
           });
-          vp.should.have.property('@context');
+          vp.should.have.property('@context',
+            'Failed to respond with a VP with `@context` intact.');
           await assert.rejects(endpoints.verifyVp(
-            require('./input/presentation-no-context-fail.json')));
+            require('./input/presentation-no-context-fail.json')),
+          'Failed to reject a VP withtout an `@context`.');
```
Comment (TallTed) [inline tests/10-vcdm2.js:108]:
```suggestion
          'Failed to reject a VP without an `@context`.');
```

File: tests/10-vcdm2.js [hunk 2]
```diff
@@ -85,10 +87,12 @@ describe('Contexts', function() {
...
           const vc = await endpoints.issue(require(
             './input/credential-ok.json'));
-          vc.should.have.property('@context');
+          vc.should.have.property('@context',
+            'Failed to respond with a VC with `@context` intact.');
           // negative @context test
           await assert.rejects(endpoints.issue(
-            require('./input/credential-no-context-fail.json')));
+            require('./input/credential-no-context-fail.json')),
+          'Failed to reject a VC withtout an `@context`.');
```
Comment (TallTed) [inline tests/10-vcdm2.js:95]:
```suggestion
          'Failed to reject a VC without an `@context`.');
```

File: tests/10-vcdm2.js [hunk 3]
```diff
@@ -85,10 +87,12 @@ describe('Contexts', function() {
           // positive @context test
           const vc = await endpoints.issue(require(
             './input/credential-ok.json'));
-          vc.should.have.property('@context');
+          vc.should.have.property('@context',
+            'Failed to respond with a VC with `@context` intact.');
```
Comment (TallTed) [inline tests/10-vcdm2.js:91]:
```suggestion
            'Failed to respond with a VC with intact `@context`.');
```

File: tests/10-vcdm2.js [hunk 4]
```diff
@@ -97,21 +101,27 @@ describe('Contexts', function() {
             presentation: require('./input/presentation-ok.json'),
             options: createOptions
           });
-          vp.should.have.property('@context');
+          vp.should.have.property('@context',
+            'Failed to respond with a VP with `@context` intact.');
```
Comment (TallTed) [inline tests/10-vcdm2.js:105]:
```suggestion
            'Failed to respond with a VP with intact `@context`.');
```

**Review by TallTed — CHANGES_REQUESTED** [review]
Fix typos and grammar

File: tests/10-vcdm2.js (hunk 3 above)
Comment (BigBlueHat) [reply tests/10-vcdm2.js:91]: This one's subjective, but if it makes you happy. 😉

**Review by TallTed — APPROVED** [review]
So far, so good

**Review by aljones15 — APPROVED** [review]
I did not know that https://nodejs.org/api/assert.html#strict-assertion-mode had an optional message. I'm assuming this has been tested and the error message shows up in the report. good work here.

**Pull request merged**